app.py

```diff
@@ -1,36 +1,88 @@
 import streamlit as st
 from pathlib import Path
 import json
+import os
+import shutil
+
 # For some reason the windows version only works if this is imported here
 import pyopenms
 
+# Prevent Streamlit from prompting for email on first run
+try:
+    import streamlit.runtime.credentials as stcreds
+
+    # Get the path to the Streamlit credentials file
+    credentials_file = Path(stcreds.get_credentials_file_path())
+
+    # Create the directory if it doesn't exist
+    os.makedirs(credentials_file.parent, exist_ok=True)
+
+    # Check if the credentials file exists
+    if not credentials_file.exists():
+        # Copy our pre-configured credentials file
+        project_root = Path(__file__).parent
+        source_credentials = project_root / ".streamlit" / "credentials.toml"
+        if source_credentials.exists():
+            shutil.copyfile(source_credentials, credentials_file)
+except Exception as e:
+    # Silently continue if there's an error
```

Comment:
This is bad practice. On my site an error occurs so the above code never runs as `get_credentials_file_path()` does not exist. Also it looks like this is executed after Streamlit prompts the user for their email.